src/Models/UserIdentityModel.php

```diff
@@ -329,13 +329,53 @@ public function revokeAllAccessTokens(User $user): void
         $this->checkQueryReturn($return);
     }
 
+    /**
+     * Force password reset for multiple users.
+     *
+     * @param int[]|string[] $userIds
+     *
+     * @return mixed
+     */
+    public function forceMultiplePasswordReset(array $userIds)
+    {
+        $this->where(['type' => Session::ID_TYPE_EMAIL_PASSWORD, 'force_reset' => 0]);
+        $this->whereIn('user_id', $userIds);
+        $this->set('force_reset', 1);
+
+        return $this->update();
+    }
+
+    /**
+     * Force global password reset.
+     * This is useful for enforcing a password reset
+     * for ALL users incase of a security breach.
+     *
+     * The logged in user's ID is ommitted as it is
+     * assummed that this is the admin (or superadmin)
+     * account.
+     *
+     * @return mixed
+     */
+    public function forceGlobalPasswordReset()
+    {
+        $whereFilter = [
+            'user_id !='  => user_id(),
```

Comment:
Please remove `user_id()`. It makes unnecessary dependency to `Auth`.
It is not permitted in deptrac config:
https://github.com/codeigniter4/shield/blob/develop/deptrac.yaml#L109-L115

If you want to exclude a user id, add a parameter like`User $userToExclude`.

Reply:
The idea I had for the `forceGlobalPasswordReset()` method was for it to automatically pick up the ID of the currently logged in user, instead of giving the option to enter it manually. 

If it is going to be added as a param, then another user ID can be passed to it, which is not the ID of the currently logged in user. Is there a better way to use the logged in user's ID?

Reply:
> If it is going to be added as a param, then another user ID can be passed to it, which is not the ID of the currently logged in user. 

It is better to pass `user_id()` value to `forceGlobalPasswordReset()`.
`UserIdentityModel` does not need to know who is logged in.

Classes should be designed to be loosely coupled to avoid needless dependencies.

Or, I don't think the feature to exclude the logged in user is necessary. 
Why exclude only myself from password reset? 
If password data has been compromised, everyone should reset password.

Reply:
> Or, I don't think the feature to exclude the logged in user is necessary.
> Why exclude only myself from password reset?
> If password data has been compromised, everyone should reset password.

Well, I felt that as an admin, the user must have manually reset his/her password before forcing password reset on all other users. But I think you have a point, forcing global password reset should apply to everyone.
I'll make the required changes asap.